Core/PARStore.m

```diff
@@ -1407,15 +1443,36 @@ - (BOOL)deleteBlobAtPath:(NSString *)path error:(NSError **)error
     // otherwise blobs are stored in a special blob directory
     __block NSError *localError = nil;
     NSURL *fileURL = [[self blobDirectoryURL] URLByAppendingPathComponent:path];
+    NSURL *tombstoneURL = [fileURL URLByAppendingPathExtension:TombstoneFileExtension];
+
     NSError *coordinatorError = nil;
-    [[self newFileCoordinator] coordinateWritingItemAtURL:fileURL options:NSFileCoordinatorWritingForReplacing error:&coordinatorError byAccessor:^(NSURL *newURL)
+    
+    [[self newFileCoordinator] coordinateWritingItemAtURL:fileURL options:NSFileCoordinatorWritingForReplacing writingItemAtURL:tombstoneURL options:NSFileCoordinatorWritingForReplacing error:&coordinatorError byAccessor:^(NSURL * _Nonnull newURL, NSURL * _Nonnull newTombstoneURL)
      {
-         // write to disk (overwrite any file that was at that same path before)
-         NSError *error = nil;
-         BOOL success = [[NSFileManager defaultManager] removeItemAtURL:fileURL error:&error];
-         if (!success)
-             localError = [NSError errorWithObject:self code:__LINE__ localizedDescription:[NSString stringWithFormat:@"Could not delete data blob at path '%@'", newURL.path] underlyingError:error];
-     }];
+        // TODO: Decide how to interpret a tombstone file being present already.
+        // With the previous implementation, I think that calling delete on a file that didn't exist
+        // (or had already been deleted) returned an error.
+        // To be consistent with this behaviour, we should perhaps also return an error if we
+        // find an existing tombstone file.
+        // However, my normal inclination would be to treat the presence of the tombstone as a sign
+        // that the file has been successfully deleted and quietly return success.
```

Comment:
I think it is safer to return an error. It gives the little bit more information. If you expect to find a file, and you don't, that is probably something you want to know about.